### PR TITLE
ci: speed up OSX CI

### PR DESCRIPTION
**Pull request by mattklein123**
This should give very basic sanity coverage.

### CODE REVIEW

**Comment by mattklein123** [comment]
We might want a few more integration tests included. I don't think this would have caught the reuse port issues so I will at least add those. If there are others let me know.

File: ci/mac_ci_steps.sh [hunk 1]
```diff
@@ -42,16 +42,21 @@ fi
...
 # and test steps).
 
+DEFAULT_TEST_TARGETS=(
+  "//test/integration:integration_test"
+  "//test/integration:protocol_integration_test"
+  "//test/integration:tcp_proxy_integration_test")
+
 if [[ $# -gt 0 ]]; then
-  TEST_TARGETS=$*
+  TEST_TARGETS=("$*")
```
Comment (phlax) [inline ci/mac_ci_steps.sh:51]: i think you can just do `TEST_TARGETS=$@` here - its already an array - this converts it to a string and back again

Comment (mattklein123) [reply ci/mac_ci_steps.sh:51]: I tried and actually neither work, it seems it has to be `("$@")`. Yay bash. :)

**Review by phlax — APPROVED** [review]
i dont know too much about macos tests - but lgtm from bash pov

**Pull request merged**